src/NuGetPackageVerifier/Rules/AssemblyStrongNameRule.cs

```diff
@@ -45,22 +45,23 @@ public IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context
                             if (AssemblyHelpers.IsAssemblyManaged(assemblyPath))
                             {
                                 isManagedCode = true;
-                                var clrStrongName = (IClrStrongName)RuntimeEnvironment.GetRuntimeInterfaceAsObject(
-                                    new Guid("B79B0ACD-F5CD-409b-B5A5-A16244610B92"),
-                                    new Guid("9FD93CCF-3280-4391-B3A9-96E1CDE77C8D"));
-
-                                bool verificationForced;
-                                hresult = clrStrongName.StrongNameSignatureVerificationEx(assemblyPath, true, out verificationForced);
-                                if (hresult == 0)
-                                {
-                                    isStrongNameSigned = true;
-                                }
-
-                                var testAssembly = AssemblyName.GetAssemblyName(assemblyPath);
-                                var tokenHexString = BitConverter.ToString(testAssembly.GetPublicKeyToken()).Replace("-", "");
-                                if (_publicKeyToken.Equals(tokenHexString, StringComparison.OrdinalIgnoreCase))
+                                using (var assembly = AssemblyDefinition.ReadAssembly(assemblyPath))
                                 {
-                                    hasCorrectPublicKeyToken = true;
+                                    if (assembly.Modules.Any())
+                                    {
+                                        isStrongNameSigned = assembly.Modules.All(
+                                            module => module.Attributes.HasFlag(ModuleAttributes.StrongNameSigned));
+                                    }
+                                    else
+                                    {
+                                        throw new InvalidOperationException("The managed assembly does not contain any modules.");
```

Comment:
I was worried here about the case of satellite resource DLLs and whether or not they have at least one module, but it seems like they always do have one, so no worries here. 🙈